### PR TITLE
Fix Topk Calculation in GPTOSS

### PR DESCRIPTION
**Pull request by SKRohit**
Fixes Accuracy Issue in GPTOSS: https://github.com/vllm-project/vllm-gaudi/pull/887. Updates `apply_monolithic`  introduced in https://github.com/vllm-project/vllm-gaudi/pull/876 to handle gptoss

### CODE REVIEW

File: vllm_gaudi/ops/hpu_fused_moe.py [hunk 1]
```diff
@@ -67,9 +67,13 @@ def apply_monolithic(
...
-            topk_weights = F.softmax(router_logits, dim=1, dtype=torch.float32)
-            topk_weights, topk_ids = torch.topk(topk_weights, layer.top_k, dim=-1)
-            topk_weights /= topk_weights.sum(dim=-1, keepdim=True)
+            if self.model_type is not None and self.model_type in ["gpt_oss"]:
+                topk_weights, topk_ids = torch.topk(router_logits, layer.top_k, dim=-1)
+                topk_weights = F.softmax(topk_weights, dim=-1, dtype=torch.float32)
+            else:
+                topk_weights = F.softmax(router_logits, dim=1, dtype=torch.float32)
+                topk_weights, topk_ids = torch.topk(topk_weights, layer.top_k, dim=-1)
+                topk_weights /= topk_weights.sum(dim=-1, keepdim=True)
```
[the remark this answers is not in the document]
Comment (SKRohit) [reply vllm_gaudi/ops/hpu_fused_moe.py]: I know. Just changing dim=-1 in softmax would resolve the issue. But for some model it might be required to apply softmax on second dimension, so I did not change it.

**Comment by SKRohit** [comment]
@iboiko-habana Can we merge?

**Review by iboiko-habana — APPROVED** [review]

**Comment by jaideepsai-narayan** [comment]
Hi @iboiko-habana @SKRohit , we encountered this issue (https://github.com/vllm-project/vllm-gaudi/issues/891#issuecomment-3817657124) earlier. However, with the new branch, we are again seeing a drop in accuracy when using the Unsloth version of GPTOSS.

image
